rollup.config.js

```diff
@@ -0,0 +1,45 @@
+// plugins that we are going to use
+import babel from 'rollup-plugin-babel';
+import minify from 'rollup-plugin-babel-minify';
+import resolve from 'rollup-plugin-node-resolve';
+
+// list of plugins used during building process
+const plugins = targets => ([
+  // use Babel to transpile to ES5
+  babel({
+    // ignore node_modules/ in transpilation process
+    exclude: 'node_modules/**',
+    // ignore .babelrc (if defined) and use options defined here
+    babelrc: false,
+    // use recommended babel-preset-env without es modules enabled
+    // and with possibility to set custom targets e.g. { node: '8' }
+    presets: [['env', { modules: false, targets }]],
+    // solve a problem with spread operator transpilation https://github.com/rollup/rollup/issues/281
+    plugins: [ 'babel-plugin-transform-object-rest-spread' ],
+    // removes comments from output
+    comments: false,
+  })
+]);
+
+// packages that should be treated as external dependencies, not bundled
+const external = []; // e.g. ['axios']
+
+export default [{
+  input: 'src/index.js',
+  output: {
+    file: 'dist/index.js',
+    name: 'SmartMessenger',
+    format: 'iife',
+    sourcemap: false
+  },
+  plugins: [ resolve({ browser: true}) ]
+}, {
+  input: 'src/index.js',
+  output: {
+    file: 'dist/index.min.js',
+    name: 'SmartMessenger',
+    format: 'iife',
+    sourcemap: true
+  },
+  plugins: [ minify(), resolve({ browser: true}) ]
```

Comment:
Given that this is a library, I think it would be more valuable to have multiple build targets that support a variety of build systems, and ways of importing, e.g.:  "esm" (future and modern, supports tree-shaking, etc), "commonjs" (for Node.js), and, possibly, "umd" (for everything else).

...and let the _consumer_ handle minification?

Reply:
We only need to support browser.

Reply:
oh, you linked to a block article about ESM, in Slack.

Reply:
There are two IIFE builds, one minified and one not. Maybe we use ESM instead? The requirement is to support IE10+. There are no dependencies at run time, so I'm not sure if we need treeshaking.

Reply:
Right, you can support both legacy browsers like ie10 and "next" browsers (tree shaking will only work for ESM?) -- does not have to be one or the other.

Reply:
I think we are stuck with IIFE for IE10.

Reply:
>There are no dependencies at run time

On that note, I wonder what the following library has, that a hand-rolled implementation does not:

https://github.com/kelektiv/node-uuid

(And this has had many iterations, over the years...)

Reply:
We can replace with a library in a future PR. I think we can resolve this thread?

Reply:
sure; as a not to my future self, I'd like to understand what lines 16 & 17 are for, here:
https://github.com/kelektiv/node-uuid/blob/master/v4.js